helios-services/src/main/java/com/spotify/helios/master/ZooKeeperMasterModel.java

```diff
@@ -622,6 +623,51 @@ public void rollingUpdateStep() {
     }
   }
 
+  private RollingUpdateOp rollingUpdateTimedoutError(final RollingUpdateOpFactory opFactory,
+                                                     final String host,
+                                                     final JobId jobId,
+                                                     final TaskStatus taskStatus) {
+    final List<TaskStatus.State> previousJobStates = getPreviousJobStates(jobId, host, 10);
+    final String previousJobStatesString = Joiner.on("->").join(previousJobStates);
+
+    final Map<String, Object> metadata = Maps.newHashMap();
+    metadata.put("jobState", taskStatus.getState());
+    metadata.put("previousJobStates", previousJobStates);
+    metadata.put("throttleState", taskStatus.getThrottled());
+
+    if (taskStatus.getThrottled().equals(ThrottleState.IMAGE_MISSING)) {
+      return opFactory.error(
+              "timed out waiting for job to reach RUNNING due to missing Docker image " +
+                      String.format("(previous job states: %s)", previousJobStatesString),
```

Comment:
instead of concatenating two strings this could also just be `String.format("timed out waiting for ... (previous job states: %s)", previousJobStatesString)`